demos/LowAllocationWebServer/IFormatterHttpExtensions.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Text.Formatting;
+
+namespace System.Net.Http.Buffered
+{
+    public static class IFormatterHttpExtensions
+    {
+        private const string Http = "HTTP/";
+        private const int ULongMaxValueNumberOfCharacters = 20;
+
+        public static void WriteHttpStatusLine<TFormatter>(
+            this TFormatter formatter, 
+            string version, 
+            string statusCode,
+            string reasonCode) where TFormatter : IFormatter
+        {
+            formatter.Append(Http);
+            formatter.Append(version);
+            formatter.AppendWhiteSpace();
+            formatter.Append(statusCode);
+            formatter.AppendWhiteSpace();
+            formatter.Append(reasonCode);
+            formatter.AppendNewLine();
+        }
+
+        public static HttpHeaderBuffer WriteHttpHeader<TFormatter>(
+            this TFormatter formatter, 
+            string name, 
+            string value,
+            int reserve = ULongMaxValueNumberOfCharacters)
+            where TFormatter : IFormatter
+        {
+            formatter.Append(name);
+            formatter.Append(" : ");
+            
+            int bytesWritten;
+            while (!value.TryFormat(formatter.FreeBuffer, default(Format.Parsed), formatter.FormattingData, out bytesWritten))
+            {                
+                formatter.ResizeBuffer();
+                bytesWritten = 0;
+            }
```

Comment:
Because for I need to get the free buffer before we commit the bytes written, so that I can hold it in a separate ByteSpan and use it to update the header value later. So that we can update the content length, for instance.